vega_sim/api/market.py

```diff
@@ -124,6 +126,8 @@ def load(self, opt: Optional[str] = None):
         self.quadratic_slippage_factor = str(config["quadratic_slippage_factor"])
         self.metadata = config["metadata"]
 
+        self.successor = Successor(opt=config["successor"])
```

Comment:
I think since `successor` is an optional field in the `NewMarketConfiguration` we should be able to handle `None` values here:
```suggestion
        self.successor = Successor(opt=config["successor"]) if config["successor"] is not None else None
```